File: 35.search-insert-position.cpp
/*
 * @lc app=leetcode id=35 lang=cpp
 *
 * [35] Search Insert Position
 */

// @lc code=start
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size()-1;
        int mid;
        while (left <= right)
        {
            mid = left + (right - left)/2;
            if (nums[mid] == target) return mid;

            if (nums[mid] < target) left = mid+1;
            else right = mid-1;
        }
        if (nums[mid] < target)
        {
            return mid+1;
        }
        else
        {
            return mid;
        }
    }
};
// @lc code=end

File: 35.search-insert-position.py
#
# @lc app=leetcode id=35 lang=python3
#
# [35] Search Insert Position
#

# @lc code=start
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        #模板2
        if target > nums[-1]:
            return len(nums)
        if target <= nums[0]:
            return 0
        left = 0
        right = len(nums)
        while left < right:
            mid = left + (right -left)//2
            if nums[mid-1] < target <= nums[mid] :
                return  mid
            if nums[mid] < target:
                left = mid + 1
            else:
                right = mid

        return None

        
# @lc code=end

